src/containers/TaskRuns/TaskRuns.js

```diff
@@ -50,14 +51,288 @@ import {
 } from '../../reducers';
 import { cancelTaskRun, deleteTaskRun, rerunTaskRun } from '../../api';
 
+const { CLUSTER_TASK, TASK } = labels;
+
+/* istanbul ignore next */
+function TaskRuns(props) {
+  const {
+    error,
+    fetchTaskRuns,
+    filters,
+    history,
+    intl,
+    isReadOnly,
+    kind,
+    loading,
+    namespace,
+    setStatusFilter,
+    statusFilter,
+    taskName,
+    taskRuns
+  } = props;
+
+  const [deleteError, setDeleteError] = useState(null);
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [toBeDeleted, setToBeDeleted] = useState([]);
+
+  useEffect(() => {
+    document.title = getTitle({ page: 'TaskRuns' });
+  }, []);
+
+  useEffect(() => {
+    reset();
+    fetchData();
+  }, [JSON.stringify(filters), namespace]);
+
+  useWebSocketReconnected(fetchData, namespace);
+
+  function fetchData() {
+    if (kind === 'ClusterTask') {
+      // TaskRuns from ClusterTask should have label 'tekton.dev/clusterTask=',
+      // (and that is the filter on the page), but some taskruns might still
+      // only have the old label 'tekton.dev/task='
+      // So, for ClusterTasks, also fetch with the old filter:
+      fetchTaskRuns({
+        filters: [`${TASK}=${taskName}`]
+      });
```

Comment:
@AlanGreene I've seen in previous implementation [here](https://github.com/tektoncd/dashboard/blob/main/src/containers/TaskRuns/TaskRuns.js#L258). It looks like firing `fetchTaskRuns` callback twice if `kind` is ClusterTask. Is that a bug?

Reply:
That's intentional, see the comment above this code. In a previous version of Tekton Pipelines, TaskRuns of a ClusterTask had the `tekton.dev/task` label instead of `tekton.dev/clusterTask`, so this is for backwards compatibility with those older versions.

We will remove it in a future release but let's keep it for now.